src/api/web/endpoint/package.ts

```diff
@@ -66,6 +67,10 @@ function addPackageWebApi(route: Router, storage: IStorageHandler, auth: IAuth,
               }
               permissions.push(pkgCopy);
             }
+
+            if (!_.isNil(pkgCopy.dist) && !_.isNull(pkgCopy.dist.tarball)) {
```

Comment:
Sorry, perhaps I did not explain good enough, but this code has to be inside of the `if (await checkAllow(pkg.name, req.remote_user)) {` block, otherwise we will mutate all packages, we don't want that, we will retrieve only those user is allowed to see, plus, it has to happen before`permissions.push(pkgCopy);`. Hopefully now is more clear, let me know if you have questions.

Reply:
@juanpicado oh, I see